backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceEventsHandler.java

```diff
@@ -0,0 +1,211 @@
+     * Maximum number of events in store. If number of events in store hit the limit
+     * BacktraceMetrics instance will send data to Backtrace.
+     */
+    private int maximumNumberOfEvents = 350;
+
+    /**
+     * The application name
+     */
+    protected String application;
+
+    /**
+     * The application version
+     */
+    protected String appVersion;
+
+    /**
+     * Create BacktraceEventsHandler instance
+     *
+     * @param context
+     * @param api
+     * @param backtraceHandlerThread
+     * @param urlPrefix
+     * @param settings
+     */
+    public BacktraceEventsHandler(Context context, Map<String, Object> customAttributes,
+                                  Api api, final BacktraceHandlerThread backtraceHandlerThread, String urlPrefix, BacktraceMetricsSettings settings) {
+        // This should always have a nonnull looper because BacktraceHandlerThread starts in the
+        // constructor and getLooper blocks until the looper is ready if the thread is started
+        //
+        // We cannot cache the looper here since the super call has to go first (i.e: before
+        // declaring and assigning a Looper variable)
+        super(backtraceHandlerThread.getLooper());
+        if (backtraceHandlerThread.getLooper() == null) {
+            throw new NullPointerException("Expected nonnull looper, this should not happen");
+        }
+        this.context = context;
+        this.customAttributes = customAttributes;
+        this.backtraceHandlerThread = backtraceHandlerThread;
+        this.api = api;
+        this.submissionUrl = settings.getBaseUrl() + "/" + urlPrefix + "/submit?token=" + settings.getToken() + "&universe=" + settings.getUniverseName();
+        this.timeBetweenRetriesMillis = settings.getTimeBetweenRetriesMillis();
+
+        long timeIntervalMillis = settings.getTimeIntervalMillis();
+
+        BacktraceAttributes backtraceAttributes = new BacktraceAttributes(context, null, null);
+        this.application = backtraceAttributes.getApplicationName();
+        this.appVersion = backtraceAttributes.getApplicationVersionOrEmpty();
+
+        if (timeIntervalMillis != 0) {
+            final BacktraceEventsHandler handler = this;
+            handler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    handler.send();
+                    handler.postDelayed(this, timeIntervalMillis);
+                }
+            }, timeIntervalMillis);
+        }
+    }
+
+    /**
+     * Number of events in the queue
+     */
+    public int getCount() {
+        return events.size();
+    }
+
+    public int getMaximumNumberOfEvents() {
+        return this.maximumNumberOfEvents;
+    }
+
+    public void setMaximumNumberOfEvents(int maximumNumberOfEvents) {
+        this.maximumNumberOfEvents = maximumNumberOfEvents;
+    }
+
+    public abstract void sendStartupEvent(String eventName);
+
+    public void send() {
+        if (events == null) {
```

Comment:
maybe it is worth to combine this `if` and this below

Reply:
Good point, will do!